terra-jupyter-dev-base/Dockerfile

```diff
@@ -0,0 +1,163 @@
+    CONDA_AUTO_UPDATE_CONDA=false \
+    CONDA_DIR=/opt/conda
+ENV HOME=/home/$USER
+    # When using PIP_USER=true packages are installed into Python site.USER_BASE, which is '/home/jupyter' for this system.
+    # Append '/home/jupyter/.local/bin' to PATH
+    # pip docs: https://pip.pypa.io/en/stable/reference/pip_install/#cmdoption-user
+ENV PATH="${PATH}:${CONDA_DIR}/bin:${HOME}/.local/bin:${HOME}/packages/bin"
+
+# Users
+RUN useradd -m -s /bin/bash $USER \
+ && usermod -g users $USER \
+ && useradd -m -s /bin/bash -N -u $WELDER_UID $WELDER_USER \
+# Prerequisites
+ && apt-get update && apt-get install -yq --no-install-recommends \
+    sudo \
+ && sudo -i \
+    echo "deb http://security.ubuntu.com/ubuntu/ bionic main" >> /etc/apt/sources.list \
+    sudo apt update libexempi3 \
+ && sudo -i \
+    echo "deb http://us.archive.ubuntu.com/ubuntu/ bionic universe" >> /etc/apt/sources.list \
+    sudo apt update libv8-3.14-dev \
+ && apt-get update && apt-get install -yq --no-install-recommends \
+    # gnupg requirement
+    dirmngr \
+    gnupg \
+    # curl requirement
+    curl \
+    ca-certificates \
+    # useful utilities for debugging within the docker
+    nano \
+    procps \
+    # extras \
+    wget \
+    bzip2 \
+    # install script requirements
+    locales \
+    # for ssh-agent and ssh-add
+    keychain \
+    # git
+    git \
+ # Uncomment en_US.UTF-8 for inclusion in generation
+ && sed -i 's/^# *\(en_US.UTF-8\)/\1/' /etc/locale.gen \
+ # Generate locale
+ && locale-gen \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/* \
+# install miniconda to $CONDA_DIR
+ && curl -so $HOME/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-py37_4.12.0-Linux-x86_64.sh \
+ && chmod +x $HOME/miniconda.sh \
+ && $HOME/miniconda.sh -b -p $CONDA_DIR \
+ && rm $HOME/miniconda.sh \
+# slim install of python packages that have a gcc dependency (cleanup included)
+ && apt-get update && apt-get install -y --no-install-recommends \
+    checkinstall \
+    build-essential \
+    zlib1g-dev \
+    # pip requirements
+    libssl-dev \
+    libbz2-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    llvm \
+    libncurses5-dev \
+    libncursesw5-dev \
+    tk-dev \
+    libffi-dev \
+    liblzma-dev \
+    python-openssl \
+    libexempi3 \
+    libv8-3.14-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && pip3 install firecloud  \
+ && pip3 install terra-notebook-utils \
+ && apt-get purge -y --auto-remove \
+    checkinstall \
+    build-essential \
+    zlib1g-dev \
+    # pip requirements
+    libssl-dev \
+    libbz2-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    llvm \
+    libncurses5-dev \
+    libncursesw5-dev \
+    tk-dev \
+    libffi-dev \
+    liblzma-dev \
+    python-openssl \
+    libexempi3 \
+    libv8-3.14-dev \
+# Install jupyter and some necessary python packages
+ && pip3 -V \
+ # For gcloud alpha storage support.
+ && pip3 install google-crc32c --target /usr/lib/google-cloud-sdk/lib/third_party \
```

Comment:
I'm a little concerned about mixing pip and conda, but I know this existed before this PR.